library/hls/src/main/java/com/google/android/exoplayer2/source/hls/playlist/HlsPlaylistParser.java

```diff
@@ -550,6 +575,9 @@ private static HlsMediaPlaylist parseMediaPlaylist(
     long segmentStartTimeUs = 0;
     long segmentByteRangeOffset = 0;
     long segmentByteRangeLength = C.LENGTH_UNSET;
+    boolean isExplicitInitSegment = false;   // Was declared (TAG_INIT_SEGMENT) vs implied.
```

Comment:
Can you describe the purpose of this variable? The whole code around initialisation segment is not very clear to me.

Reply:
Sure..  The Apple sample iFrame playlists do not have EXT-X-MAP for their byte-range segments, so the Media Initialization Segment is "implicit",  one sentence in the Pantos draft explains this (I missed it first pass too).  This commit's 1f11233ba022e00367ed32904092070d8c42f5ee comment explains it.  Their sample streams are authored this way, `mediafilesegmenter` does this.  IMHO, it is a hack, but it is their 'standard'.